core/src/main/java/io/grpc/ChannelImpl.java

```diff
@@ -415,4 +458,29 @@ public void close(ScheduledExecutorService instance) {
           instance.shutdown();
         }
       };
+
+  private static final class InactiveTransport implements ClientTransport {
+    private static final InactiveTransport INSTANCE = new InactiveTransport();
+
+    @Override
+    public ClientStream newStream(
+        MethodDescriptor<?, ?> method, Headers headers, ClientStreamListener listener) {
+      throw new IllegalStateException("Transport is inactive");
```

Comment:
This should call `close()` on the `listener` and return a `NoopClientStream`. The status should be the one passed to `transportShutdown()`